VKT(Frontend): OPHKIOS-92 UI-parannuksia

pkoivisto opened the pull request.

pkoivisto committed:
VKT(Frontend): Improve front page legibility

pkoivisto committed:
VKT(Frontend): Drop pagination from exam listing table

pkoivisto committed:
VKT(Frontend): File upload instructions as a bullet point list [deploy]

pkoivisto committed:
VKT(Frontend): Fix DOM nesting

pkoivisto committed:
VKT(Frontend): Separate step for education details, move address details to same step as other contact details [deploy]

pkoivisto committed:
VKT(Frontend): Remove now obsolete privacy policy page

pkoivisto committed:
VKT(Frontend): Fix layout for education details

pkoivisto committed:
VKT(Frontend): Bump Cypress version

pkoivisto committed:
VKT(Frontend): Fix tests [deploy]

pkoivisto committed:
SHARED(Frontend): Include a file drop zone by default on desktop with the FileUpload component

pkoivisto committed:
VKT(Frontend): Take updated FileUpload component into use [deploy]

pkoivisto committed:
VKT(Frontend): Minor fixes [deploy]

pkoivisto committed:
VKT(Frontend): Try to fix focusing on stepper heading

pkoivisto committed:
VKT(Frontend): Add dividers between sections of preview page, move address details together with other contact details [deploy]

pkoivisto committed:
VKT(Frontend): Fix translation string [deploy]

pkoivisto committed:
VKT(Frontend): Show UploadAttachments also when user has run out of free enrollments

pkoivisto committed:
VKT(Frontend): Fix error handling when using fetch to POST instead of axios [deploy]

jrkkp commented:
Testailin tätä pallerolla joten en ole ihan varma onko nämä tästä johtuvia glitchejä, mutta lokaalisti hankala luurilla testata. Kaikki vaikuttas toimivan, mut jotain leiskaan liittyviä pikku juttuja sattui silmään.

<img width="419" alt="Näyttökuva 2024-09-05 kello 16 58 07" src="https://github.com/user-attachments/assets/f626db4b-6007-41a1-9423-75f063608411">
<img width="434" alt="Näyttökuva 2024-09-05 kello 16 57 58" src="https://github.com/user-attachments/assets/aa244b4d-b6d0-4680-9b29-df50eba9f2e5">
<img width="419" alt="Näyttökuva 2024-09-05 kello 16 57 25" src="https://github.com/user-attachments/assets/1b7ef261-b4a1-48bd-afc1-0beed6e7913b">

pkoivisto committed:
VKT(Frontend): Improve enrollment preview step layout

pkoivisto committed:
VKT(Frontend): Fix overflow of privacy checkbox label on mobile

pkoivisto committed:
VKT(Frontend): Allow step contents to grow more on phone, add some more padding as a hack [deploy]

lket committed:
VKT(Backend): Fix saving files to localstack on Linux [deploy]

Instant.now() may set time with nanosecond precision, depending on
platform. At least the Localstack S3 implementation requires
pre-signed url expiration timestamp precision be six null-padded
microsecond digits.

Overwriting expiration timestamp microsecond part to 000001 fixes this
problem.

Localstack implementation is:

POLICY_EXPIRATION_FORMAT2 = "%Y-%m-%dT%H:%M:%S.%fZ"
https://github.com/localstack/localstack/blob/0b4846bc9b28247a2156fce40fcdd61da74b9d76/localstack-core/localstack/services/s3/presigned_url.py#L87

pkoivisto committed:
VKT(Backend): Add missing imports [deploy]

pkoivisto committed:
VKT(Frontend): Rework enrollment preview structure on mobile to better match designs [deploy]

pkoivisto commented:
> Testailin tätä pallerolla joten en ole ihan varma onko nämä tästä johtuvia glitchejä, mutta lokaalisti hankala luurilla testata. Kaikki vaikuttas toimivan, mut jotain leiskaan liittyviä pikku juttuja sattui silmään.

Koitin vähän fiksailla näitä kauneusvirheitä. Huomasin myös että mobiililla tuo esikatselunäkymä oli aika lailla erilainen kuin Figmassa, erityisesti noiden valittujen tutkintojen ja osakokeiden osalta, joten muokkasin sitä osuutta vielä uusiksi.

jrkkp reviewed (approved):
Oon tätä jonkun verran testaillu eikä löydy mitään ongelmia, niin voinee varmaan mergailla. Diffistä oli nyt jotenkin hankala hahmottaa muutoksia, niin en keksiny koodista mitään kommentoitavaa, mutta vaikuttaa toimivan joten eiköhän tämän voi hyväksyä.
